docs/book/src/architecture/controllers/cluster.md

```diff
@@ -1 +1,45 @@
-# Cluster
+# Cluster Controller
+
+The cluster controller is responsible for keeping the cluster object in sync with the infrastructure cluster object.
+
+Its main responsibilities are:
+
+* Setting an OwnerReference on the infrastructure object.
+* Deleting all related objects in the correct order.
+* Keeping the Cluster object up to date with external objects.
+* Creating a kubeconfig secret for child clusters.
+
+## Contracts
+
+### Infrastructure Provider
+
+The InfrastructureCluster object **must** have a status field.
+
+In the status field it must have:
+
+* `ready` - a boolean field
+* `apiEndpoints` - a slice of strings
+
+Example:
+```yaml
+kind: MyCluster
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+status:
+    ready: true
+    apiEndpoints:
+    - example.com:3333
+    - example.com:3334
+```
+
+### Secrets
+
+| secret name | field name | content |
+|:---:|:---:|:---:|
+|`<cluster-name>-ca`|`tls.crt`|base64 encoded TLS certificate in PEM format|
+|`<cluster-name>-ca`|`tls.key`|base64 encoded TLS private key in PEM format|
+
+OR
```

Comment:
This is a little confusing, we should specify when you'd want one or the other, wdyt?

Reply:
yeah, I'll add some words about each method